lib/NativeCode/DynamicLibraryLoaderHelper/NativeRender/src/eos_helpers.cpp

```diff
@@ -190,223 +184,249 @@ namespace pew::eos
-            }
-            else
-            {
-                // steam_platform needs to have a count of how many bytes the "array" is, stored in SteamApiInterfaceVersionsArrayBytes
-                // This has some fuzzy behavior; if you set it to 0 or count it up properly, there won't be a logged problem
-                // if you put a non-zero amount that is insufficient, there will be an unclear logged error message
-                steam_platform.SteamApiInterfaceVersionsArrayBytes = static_cast<uint32_t>(size);
-            }
+        platform_options.EncryptionKey = platform_config.client_credentials.encryption_key.c_str();
 
-            steam_integrated_platform_option.ApiVersion = EOS_INTEGRATEDPLATFORM_OPTIONS_API_LATEST;
-            steam_integrated_platform_option.Type = EOS_IPT_Steam;
-            steam_integrated_platform_option.Flags = eos_steam_config.flags;
-            steam_integrated_platform_option.InitOptions = &steam_platform;
+        platform_options.OverrideCountryCode = platform_config.overrideCountryCode.empty() ? nullptr : platform_config.overrideCountryCode.c_str();
+        platform_options.OverrideLocaleCode = platform_config.overrideLocaleCode.empty() ? nullptr : platform_config.overrideLocaleCode.c_str();
 
-            steam_platform.ApiVersion = EOS_INTEGRATEDPLATFORM_STEAM_OPTIONS_API_LATEST;
+        platform_options.ProductId = product_config.product_id.c_str();
+        platform_options.SandboxId = platform_config.deployment.sandbox.id.c_str();
+        platform_options.DeploymentId = platform_config.deployment.id.c_str();
+        platform_options.ClientCredentials.ClientId = platform_config.client_credentials.client_id.c_str();
+        platform_options.ClientCredentials.ClientSecret = platform_config.client_credentials.client_secret.c_str();
 
-            EOS_IntegratedPlatform_CreateIntegratedPlatformOptionsContainerOptions options = { EOS_INTEGRATEDPLATFORM_CREATEINTEGRATEDPLATFORMOPTIONSCONTAINER_API_LATEST };
-            eos_library_helpers::EOS_IntegratedPlatform_CreateIntegratedPlatformOptionsContainer_ptr(&options, &integrated_platform_options_container);
-            platform_options.IntegratedPlatformOptionsContainerHandle = integrated_platform_options_container;
+        platform_options.TickBudgetInMilliseconds = platform_config.tick_budget_in_milliseconds;
 
-            EOS_IntegratedPlatformOptionsContainer_AddOptions addOptions = { EOS_INTEGRATEDPLATFORMOPTIONSCONTAINER_ADD_API_LATEST };
-            addOptions.Options = &steam_integrated_platform_option;
-            eos_library_helpers::EOS_IntegratedPlatformOptionsContainer_Add_ptr(integrated_platform_options_container, &addOptions);
+        if (platform_config.task_network_timeout_seconds > 0)
+        {
+            double task_network_timeout_seconds_dbl = platform_config.task_network_timeout_seconds;
+            platform_options.TaskNetworkTimeoutSeconds = &task_network_timeout_seconds_dbl;
```

Comment:
This also is reference to a stack value that won't exist outside of this call.

Reply:
I fixed this to call new - and added proper `delete` calls when `platform_options` is no longer needed.

Reply:
I mean line 371 creates a pointer to a stack value, `task_network_timeout_seconds_dbl`, which won't survive outside of this `if` statement.

Reply:
I believe that it instantiates a new block of memory and uses a copy operator to copy the value from `platform_config.task_network_timeout_seconds`. Is that not the case?

Reply:
Testing confirms my assumption about the persistence of the memory.

Reply:
line 370 copies the value from from `platform_config` and stores it in a double that exists on the stack for just frame. i.e. the value `&task_network_timeout_seconds_dbl`  refers to a stack address, however that is only valid till this if statement terminates. On most C++ implementations (that I've seen, at least), it will probably live till at least line `378`; however once something needs more stack space, i.e. the calling function calls another function that needs more stack space than it did previously, the value stored at that address could get overwritten.

Reply:
I think I see the confusion - this comment thread is showing outdated implementation - but because the comment thread was started on that version of the implementation - it still shows it for context.

The updated implementation can be seen here:

https://github.com/PlayEveryWare/eos_plugin_for_unity/blob/c7b4dd3f197df212a60cf1e28ce9b5e23ec4291e/lib/NativeCode/DynamicLibraryLoaderHelper/NativeRender/src/eos_helpers.cpp#L368-372

Reply:
Ah okay I see. I'll accept this for now but in the future we should work to change the way init is called so we don't need to allocate any memory.